pkg/shed/index.go

```diff
@@ -162,32 +161,37 @@ func (f Index) Get(keyFields Item) (out Item, err error) {
 // fields. Every item must have all fields needed for encoding the
 // key set. The passed slice items will be changed so that they
 // contain data from the index values. No new slice is allocated.
-// This function uses a single leveldb snapshot.
 func (f Index) Fill(items []Item) (err error) {
-	snapshot, err := f.db.ldb.GetSnapshot()
-	if err != nil {
-		f.logger.Debugf("error getting snapshot in Fill. Error: %s", err.Error())
-		return err
-	}
-	defer snapshot.Release()
-
+	txn := f.db.GetBatch(false)
+	defer txn.Discard()
 	for i, item := range items {
 		key, err := f.encodeKeyFunc(item)
 		if err != nil {
 			f.logger.Debugf("keyfields encoding error in Fill. Error: %s", err.Error())
 			return err
 		}
-		value, err := snapshot.Get(key, nil)
+		badgerItem, err := txn.Get(key)
 		if err != nil {
+			if err == badger.ErrKeyNotFound {
+				f.logger.Debugf("error could not find key in fill. Error: %s", err.Error())
+				return ErrNotFound
+			}
 			f.logger.Debugf("error getting key %s in Fill. Error: %s", string(key), err.Error())
 			return err
 		}
-		v, err := f.decodeValueFunc(item, value)
+		var decodedItem Item
+		err = badgerItem.Value(func(val []byte) error {
+			decodedItem, err = f.decodeValueFunc(item, val)
+			if err != nil {
+				return err
+			}
+			return nil
```

Comment:
Can be `return err`.

Reply:
i think it is correct... it should be return nil

Reply:
The if is not needed, the comment is meant for all 4 lines, including if.